Added support for setting up release version

schaefi opened the pull request:
Currently the release version is not set or set to '0' for package managers which requires a value to operate. However, in order to support leveraging the same description across different releases it is important to have the opportunity to specify a setting for the release version. This commit adds a new optional attribute to the preferences section which allows to specify a custom value which serves as the release version:

```xml
<preferences>
    <release_version>TEXT</release_version>
</preferences>
```

If not specified the default setting as before applies. Please note the release version information is currently used in dnf and microdnf package managers only. It might happen that it gets applied to the other package manager backends as well. This will happen on demand though.

Related to Issue #1918. This Fixes #1927

schaefi commented:
@Conan-Kudo if you look at this, please take a special look on the change in the ```dnf.py``` code. We only explicitly set --releasever in microdnf and this change here also applies the setting to dnf. I think I did that correctly but I haven't tested it. The integration tests will outline issues but maybe we can double check before

Conan-Kudo commented:
I will test this later today and give the appropriate feedback. 😉

schaefi commented:
@Conan-Kudo any news on this one ?

Conan-Kudo commented:
I get the following error with this patch:

```
[root@a75839102358 kiwi-desc]# kiwi-ng --type=oem --profile=OpenStack system build --description ./centos9 --target-dir $PWD/tmpoutput
[ INFO    ]: 14:59:24 | Reading runtime config file: '/etc/kiwi.yml'
[ INFO    ]: 14:59:24 | Loading XML description
[ INFO    ]: 14:59:24 | Support for multiple markup descriptions available
[ WARNING ]: 14:59:24 | Command "jing" not found in the environment
[ WARNING ]: 14:59:24 | For detailed schema validation report, install: jing
[ INFO    ]: 14:59:24 | Showing only raw library error log:
[ INFO    ]: 14:59:24 | --> /var/tmp/kiwi_xslt-6al3mple:14:0:ERROR:RELAXNGV:RELAXNG_ERR_EXTRACONTENT: Element preferences has extra content: release_version
[ ERROR   ]: 14:59:24 | KiwiDescriptionInvalid: Failed to validate schema and/or schematron rules. Use --debug for more details
```

schaefi commented:
```
div {
    k.release-version.content = text
```

The element is named ```<release-version>``` not ```<release_version>```

I check if I did a mistake in the docs

schaefi commented:
... make me realize I forgot to add the docs, shame on me.

schaefi commented:
@Conan-Kudo, I've updated the PR and added the missing doc hint. Also fixed the commit message which uses the wrong section name. If you adapt your example and change '_' to '-' your XML will be valid. As a side note, if you have jing installed you get way better error messages as the ones from libXML

Conan-Kudo commented:
@schaefi It's not being passed in for bootstrap, which causes it to fail.

Conan-Kudo reviewed (approved):
It works now!

schaefi commented:
@Conan-Kudo great, thanks much for testing. I'll merge and make a Staging update to see if all tests pass
